### ISSUE TITLE
Build failure on 32-bit x86 on Fedora Rawhide (Fedora 42)

### ISSUE THREAD

**Issue by Conan-Kudo** [any1/neatvnc#132]
I was trying to update neatvnc to 0.9.0 in Fedora Rawhide (GCC 14) and I got this build failure on i686:

```
[3/44] gcc -Ilibneatvnc.so.0.0.0.p -I. -I.. -I../include -I/usr/include/pixman-1 -I/usr/include/libdrm -I/usr/include/p11-kit-1 -I/usr/include/ffmpeg -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=gnu11 '-DPROJECT_VERSION="0.9.0"' -D_GNU_SOURCE -fvisibility=hidden -DAML_UNSTABLE_API=1 -Wmissing-prototypes -Wno-unused-parameter -Wno-format-truncation -DNDEBUG -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m32 -march=i686 -mtune=generic -msse2 -mfpmath=sse -mstackrealign -fasynchronous-unwind-tables -fstack-clash-protection -fPIC -DWITH_GZFILEOP -MD -MQ libneatvnc.so.0.0.0.p/src_server.c.o -MF libneatvnc.so.0.0.0.p/src_server.c.o.d -o libneatvnc.so.0.0.0.p/src_server.c.o -c ../src/server.c
FAILED: libneatvnc.so.0.0.0.p/src_server.c.o 
gcc -Ilibneatvnc.so.0.0.0.p -I. -I.. -I../include -I/usr/include/pixman-1 -I/usr/include/libdrm -I/usr/include/p11-kit-1 -I/usr/include/ffmpeg -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=gnu11 '-DPROJECT_VERSION="0.9.0"' -D_GNU_SOURCE -fvisibility=hidden -DAML_UNSTABLE_API=1 -Wmissing-prototypes -Wno-unused-parameter -Wno-format-truncation -DNDEBUG -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m32 -march=i686 -mtune=generic -msse2 -mfpmath=sse -mstackrealign -fasynchronous-unwind-tables -fstack-clash-protection -fPIC -DWITH_GZFILEOP -MD -MQ libneatvnc.so.0.0.0.p/src_server.c.o -MF libneatvnc.so.0.0.0.p/src_server.c.o.d -o libneatvnc.so.0.0.0.p/src_server.c.o -c ../src/server.c
../src/server.c: In function ‘ext_clipboard_save_provide_msg’:
../src/server.c:1522:25: error: passing argument 2 of ‘compress’ from incompatible pointer type [-Wincompatible-pointer-types]
 1522 |                         &server->ext_clipboard_provide_msg.length,
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                         |
      |                         size_t * {aka unsigned int *}
In file included from ../include/common.h:21,
                 from ../include/fb.h:25,
                 from ../src/server.c:20:
```

**Comment by any1** [comment]
Ahh, thanks for reporting. A 32 bit CI would have caught that.

**Comment by Conan-Kudo** [comment]
This was fixed with e0e0ce5c579cafc763992f1c1bb964eb95999fb7. Thanks!
